### PR TITLE
Sat 31 aug 2019 recipe

### PR DESCRIPTION
**Pull request by BXLnightManu**
Recipe schema, including client and ingredient data.

### CODE REVIEW

File: src/models/recipe.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const { Schema, model, SchemaTypes } = require("mongoose");
+
+const BASEUNITTYTYPE = ['kg', 'l'];
+
+const schema = new Schema ({
+    name: {
+        type: String,
+        required: true,
+        unique: false
```
[the remark this answers is not in the document]
Comment (ngasst) [reply src/models/recipe.js]: Therefore, no need to redeclare them!

File: src/models/client.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const { Schema, model } = require('mongoose');
+
+const schema = new Schema ({
+    name: {
+        type: String,
+        required: true,
+        unique: false
```
Comment (ngasst) [inline src/models/client.js]: Why not unique ?

File: src/models/recipe.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (BXLnightManu) [reply src/models/recipe.js]: Je comprend pas les remarques :)

File: src/models/recipe.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    baseUnit: {
+        type: String,
+        enum: BASEUNITTYTYPE,
+        required: true,
+        unique: false
+    },
+    instructions: {
```
[the remark this answers is not in the document]
Comment (BXLnightManu) [reply src/models/recipe.js]: Comment ca il faut que ce soit un array ?

File: src/models/recipe.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            required: true,
+            unique: false
+        },
+        stepDescription: {
+            type: String,
+            required: true,
+            unique: false
+        }
+    },
+    ingredientId: {
```
[the remark this answers is not in the document]
Comment (BXLnightManu) [reply src/models/recipe.js]: ok

File: src/models/recipe.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        type: SchemaTypes.ObjectId,
+        ref: 'Ingredient'
+    },
+    clientId: {
```
[the remark this answers is not in the document]
Comment (BXLnightManu) [reply src/models/recipe.js]: ok

File: src/models/recipe.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ngasst) [reply src/models/recipe.js]: Yes. It needs to be of the form:
```js
instructions: [{name: {type...}}]
```

File: src/models/recipe.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ngasst) [reply src/models/recipe.js]: See above

**Pull request merged**